packages/core/src/App/Components/Layout/Header/wallets/account-info-wallets.tsx

```diff
@@ -127,14 +130,19 @@ const AccountInfoWallets = observer(({ is_dialog_on, toggleDialog }: TAccountInf
             active_wallet.dtrade_loginid || linked_wallets_accounts.dtrade?.[0]?.loginid;
 
         // switch to dtrade account
-        if (linked_dtrade_trading_account_loginid && linked_dtrade_trading_account_loginid !== loginid) {
+        if (
+            linked_dtrade_trading_account_loginid &&
+            linked_dtrade_trading_account_loginid !== loginid &&
+            !is_wallets_cashier_route
+        ) {
             switchAccount(linked_dtrade_trading_account_loginid);
```

Comment:
avoid switching to DerivApps account when the user is on `/wallets/cashier` route